src/main/scala/org/broadinstitute/hail/driver/BuildDocs.scala

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.hail.driver
+
+import org.broadinstitute.hail.Utils._
+import org.kohsuke.args4j.{Option => Args4jOption, CmdLineParser}
+import scala.collection.JavaConverters._
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+
+object BuildDocs extends Command {
+
+  class Options extends BaseOptions {
+    @Args4jOption(required = true, name = "-o", aliases = Array("--output"), usage = "output file location")
+    var outputFile: String = _
+  }
+
+  def description = "Builds documentation"
+
+  def name = "builddocs"
+
+  override def hidden = true
+
+  def requiresVDS = false
+
+  def supportsMultiallelic = true
+
+  def newOptions = new Options
+
+  def run(state: State, options: Options): State = {
+
+    val orderedCommandList = ToplevelCommands.commands.flatMap {
+      case (s, y: SuperCommand) => Option(y.subcommands.map { case (a, b) => (s + " " + a, b) })
+      case (s, x: Command) => Option(Map[String, Command]((s, x)))
+      case _ => None
+    }.flatten.toIndexedSeq.sortWith((x1, x2) => x1._1 < x2._1)
+
+    def createSynopsis(cmdName: String, parser: CmdLineParser): String = {
+      val options = parser.getOptions.iterator().asScala
+      val sb = new StringBuilder
+
+      sb.append(cmdName)
+      options.foreach { oh =>
+        sb.append(' ')
+        if (!oh.option.required)
+          sb.append('[')
+        sb.append(oh.getNameAndMeta(null, parser.getProperties))
+        if (oh.option.isMultiValued) {
+          sb.append(" ...")
+        }
+        if (!oh.option.required())
+          sb.append(']')
+      }
+
+      sb.result()
+    }
+
+    def createCommandOptionsJSON() = {
+      pretty(JObject(orderedCommandList.map { case (s, c) =>
+        val parser = new CmdLineParser(c.newOptions)
+        val options = parser.getOptions.iterator().asScala
+        val synopsis = createSynopsis(s, parser)
+
+        JField(s, JObject(List(
+          JField("synopsis", JString(synopsis)),
```

Comment:
Then the sysnopsis variable (used only once) becomes mostly redundant and you can write `JField("synopsis", JString(synopsis(s, parser)),`.